src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php

```diff
@@ -637,6 +637,13 @@ private function annotationIsSuperfluous(
      */
     private function toComparableNames(array $types, ?string $namespace, ?string $currentSymbol, array $symbolShortNames): array
     {
+        if (1 === \count($types) && '?' === $types[0][0]) {
```

Comment:
```suggestion
        if ('?' === $types[0][0]) {
```
Checking for `?` as the first character is enough, it is not possible to mix `?` with the union type.

The first mutation worries me, why mutant with the change:
```diff
- if (1 === \count($types) && '?' === $types[0][0]) { 
+ if (1 !== \count($types) && '?' === $types[0][0]) {
```
escapes? Or it is not, and something related to Infection is not working.

Reply:
`?` check only - with current PHP yes. in future, maybe `?foo|bar` syntax would be allowed (like alias for `null|foo|bar`). dunno. but I know I won't remember to update the code, so I prefer to be double safe. no strong opinion on it.

infection - i deep dive. `toComparableNames` is called too many times, let's see if it improved for this PR after https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8132 is merged